### PR TITLE
Update dev frontend docs to Docker-first approach (#15691)

### PR DESCRIPTION
**Pull request by jzohrab**
Doc changes only to address #15691.

### CODE REVIEW

File: docs/dev/development/frontend.rst [hunk 1]
```diff
@@ -8,14 +8,35 @@ serving.
...
+    $ docker compose run --rm static npm run build
+
+
+Building outside of Docker
+--------------------------
+
+    Note: building outside of Docker is **not recommended** as it may
+    install platform-specific dependencies.
+
+Install `NodeJS 16.x <https://nodejs.org/en/download/releases/>`_,
```
Comment (miketheman) [inline docs/dev/development/frontend.rst:37]:
```suggestion
Install `NodeJS 20.x <https://nodejs.org/en/download/releases/>`_,
```

This has fallen out of sync with the actual version we build against

**Review by miketheman — COMMENTED** [review]
Excellent, thanks! One note inline.

File: docs/dev/development/frontend.rst (hunk 1 above)
Comment (jzohrab) [reply docs/dev/development/frontend.rst]: Update, thank you.

**Comment by jzohrab** [comment]
Updated and rebased off of current upstream/main.

**Pull request merged**